base/datatype_reference.bal

```diff
@@ -78,3 +78,58 @@ public type Reference record {|
     Identifier identifier?;
     string display?;
 |};
+
+# Create a relative Reference to FHIR Resource.
+#
+# + referenceResourceModel - FHIR Resource model of the Reference
+# + relativePathToReference - Relative path to Reference
+# + return - FHIRError if the Reference resource type is not defined, else FHIR Reference
+public isolated function createRelativeFhirReference(typedesc<anydata> referenceResourceModel, string relativePathToReference) returns Reference|FHIRError {
+    ResourceDefinitionRecord? def = referenceResourceModel.@ResourceDefinition;
+    string? resourceType = def?.resourceType;
+
+    if resourceType !is string {
+        return createFHIRError("Resource type is not defined", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
+    }
+    return createReference(string `${resourceType}/${relativePathToReference}`, resourceType);
+}
+
+# Create an absolute Reference to FHIR Resource.
+#
+# + referenceResourceModel - FHIR Resource model of the Reference
+# + absolutePathToReference - Absolute path to Reference
+# + return - FHIRError if the Reference resource type is not defined, else FHIR Reference
+public isolated function createAbsoluteFhirReference(typedesc<anydata> referenceResourceModel, string absolutePathToReference) returns Reference|FHIRError {
+    ResourceDefinitionRecord? def = referenceResourceModel.@ResourceDefinition;
+    string? resourceType = def?.resourceType;
+
+    if resourceType !is string {
+        return createFHIRError("Resource type is not defined", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
+    }
+    return createReference(absolutePathToReference, resourceType);
+}
+
+# Create a contained Reference to FHIR Resource.
+#
+# + targetResource - FHIR Resource which the Reference is to be contained/embedded
+# + resourceTypeReference - Resource typed Reference to be contained/embedded
+# + return - FHIRError if the target FHIR Resource is not a valid FHIR resource, else FHIR Reference
+public isolated function createContainedFhirReference(map<anydata> targetResource, Resource resourceTypeReference) returns Reference|FHIRError {
+    string resourceType = resourceTypeReference.resourceType;
+    anydata? targetContained = targetResource["contained"];
+    if targetContained is () {
+        Resource[] containedResources = [];
+        containedResources.push(resourceTypeReference);
+        targetResource["contained"] = containedResources;
+    } else if targetContained is Resource[] {
+        targetContained.push(resourceTypeReference);
+    }
+
+    string? referenceId = resourceTypeReference.id;
+    if referenceId !is string {
+        return createFHIRError("Contained reference id is missing", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
+    }
+    return createReference(string `#${referenceId}`, resourceType);
+}
+
+isolated function createReference(string reference, string resourceType) returns Reference => {reference: reference, 'type: resourceType};
```

Comment:
```suggestion
isolated function createReference(string reference, string resourceType) returns Reference => {reference: reference, 'type: resourceType};

```

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/e78b0cdc6f70182a93d96bdce43b476ba4622302